airbyte-integrations/bases/base-java/src/test/java/io/airbyte/integrations/destination_async/DetectStreamToFlushTest.java

```diff
@@ -0,0 +1,171 @@
+      final DetectStreamToFlush detectStreamToFlush = new DetectStreamToFlush(bufferDequeue, null, isClosing);
+      assertEquals(SIZE_10MB, detectStreamToFlush.computeQueueThreshold());
+    }
+
+    @Test
+    void testClosingThreshold() {
+      final AtomicBoolean isClosing = new AtomicBoolean(true);
+      final BufferDequeue bufferDequeue = mock(BufferDequeue.class);
+      final DetectStreamToFlush detectStreamToFlush = new DetectStreamToFlush(bufferDequeue, null, isClosing);
+      assertEquals(0, detectStreamToFlush.computeQueueThreshold());
+    }
+
+    @Test
+    void testEagerFlushThresholdBelowThreshold() {
+      final AtomicBoolean isClosing = new AtomicBoolean(false);
+      final BufferDequeue bufferDequeue = mock(BufferDequeue.class);
+      when(bufferDequeue.getTotalGlobalQueueSizeBytes()).thenReturn(8L);
+      when(bufferDequeue.getMaxQueueSizeBytes()).thenReturn(10L);
+      final DetectStreamToFlush detectStreamToFlush = new DetectStreamToFlush(bufferDequeue, null, isClosing);
+      assertEquals(SIZE_10MB, detectStreamToFlush.computeQueueThreshold());
+    }
+
+    @Test
+    void testEagerFlushThresholdAboveThreshold() {
+      final AtomicBoolean isClosing = new AtomicBoolean(false);
+      final BufferDequeue bufferDequeue = mock(BufferDequeue.class);
+      when(bufferDequeue.getTotalGlobalQueueSizeBytes()).thenReturn(9L);
+      when(bufferDequeue.getMaxQueueSizeBytes()).thenReturn(10L);
+      final DetectStreamToFlush detectStreamToFlush = new DetectStreamToFlush(bufferDequeue, null, isClosing);
+      assertEquals(0, detectStreamToFlush.computeQueueThreshold());
+    }
+
+  }
+
+  @Nested
+  class GetNextStreamToFlush {
+
+    @Test
+    void testGetNextSkipsEmptyStreams() {
+      final BufferDequeue bufferDequeue = mock(BufferDequeue.class);
+      when(bufferDequeue.getBufferedStreams()).thenReturn(Set.of(DESC1));
+      when(bufferDequeue.getQueueSizeBytes(DESC1)).thenReturn(Optional.of(0L));
+      final RunningFlushWorkers runningFlushWorkers = mock(RunningFlushWorkers.class);
+      when(runningFlushWorkers.getNumFlushWorkers(any())).thenReturn(0);
+      final DetectStreamToFlush detectStreamToFlush = new DetectStreamToFlush(bufferDequeue, runningFlushWorkers, new AtomicBoolean(false));
+      assertEquals(Optional.empty(), detectStreamToFlush.getNextStreamToFlush(0));
+    }
+
+    @Test
+    void testGetNextPicksUpOnSizeTrigger() {
+      final BufferDequeue bufferDequeue = mock(BufferDequeue.class);
+      when(bufferDequeue.getBufferedStreams()).thenReturn(Set.of(DESC1));
+      when(bufferDequeue.getQueueSizeBytes(DESC1)).thenReturn(Optional.of(1L));
+      final RunningFlushWorkers runningFlushWorkers = mock(RunningFlushWorkers.class);
+      when(runningFlushWorkers.getNumFlushWorkers(any())).thenReturn(0);
+      final DetectStreamToFlush detectStreamToFlush = new DetectStreamToFlush(bufferDequeue, runningFlushWorkers, new AtomicBoolean(false));
+      // if below threshold, triggers
```

Comment:
i think this is reversed? should be if above threshold, triggers if i understand correctly